pkg/fftypes/identity.go

```diff
@@ -16,9 +16,226 @@
+}
+
+// IdentityProfile are the field of a profile that can be updated over time
+type IdentityProfile struct {
+	Description string     `json:"description,omitempty"`
+	Profile     JSONObject `json:"profile,omitempty"`
+}
+
+// Identity is the persisted structure backing all identities, including orgs, nodes and custom identities
 type Identity struct {
+	IdentityBase
+	IdentityProfile
+	Messages IdentityMessages `json:"messages,omitempty"`
+	Created  *FFTime          `json:"created,omitempty"`
+	Updated  *FFTime          `json:"updated,omitempty"`
+}
+
+// IdentityCreateDTO is the input structure to submit to register an identity.
+// The blockchain key that will be used to establish the claim for the identity
+// needs to be provided.
+type IdentityCreateDTO struct {
+	Name   string       `json:"name"`
+	Type   IdentityType `json:"type,omitempty"`
+	Parent *UUID        `json:"parent,omitempty"`
+	Key    string       `json:"key,omitempty"`
+	IdentityProfile
+}
+
+// IdentityUpdateDTO is the input structure to submit to update an identityprofile.
+// The same key in the claim will be used for the update.
+type IdentityUpdateDTO struct {
+	IdentityProfile
+}
+
+// SignerRef is the nested structure representing the identity that signed a message.
+// It might comprise a resolvable by FireFly identity DID, a blockchain signing key, or both.
+type SignerRef struct {
 	Author string `json:"author,omitempty"`
 	Key    string `json:"key,omitempty"`
 }
+
+// IdentityClaim is the data payload used in a message to broadcast an intent to publish a new identity.
+// Most claims (except root orgs, where different requirements apply) require a separate IdentityValidation
+// from the parent identity to be published (on the same topic) before the identity is considered valid
+// and is stored as a confirmed identity.
+type IdentityClaim struct {
+	Identity *Identity `json:"identity"`
+}
+
+// IdentityVerification is the data payload used in message to broadcast a verification of a child identity.
+// Must refer to the UUID and Hash of the IdentityClaim message, and must contain the same base identity data.
+type IdentityVerification struct {
+	Claim    MessageRef   `json:"claim"`
+	Identity IdentityBase `json:"identity"`
+
+	// SignerRef lets us store back the message when broadcasting, but isn't part of the payload
+	SignerRef *Identity `json:"-"`
+}
+
+// IdentityUpdate is the data payload used in message to broadcast an update to an identity profile.
+// The broadcast must be on the same identity as the currently established identity claim message for the identity,
+// and it must contain the same identity data.
+// The profile is replaced in its entirety.
+type IdentityUpdate struct {
+	Identity IdentityBase    `json:"identity"`
+	Updates  IdentityProfile `json:"updates,omitempty"`
+
+	// IdentityRef lets us store back the message when broadcasting, but isn't part of the payload
+	IdentityRef *Identity `json:"-"`
```

Comment:
Slightly confused about how this is used "when broadcasting" but isn't part of the broadcast payload. I may be missing something subtle.

Reply:
Clarified comment - I got a better understanding after I wrote this one.

Reply:
Actually, might go the same route as with verifications and rip this out. I was trying for consistency, but the pattern of there being multiple messages for this definition just didn't fit in the end.

Reply:
... a future PR could go a step further and rip out the `SetBroadcastMessage()` function completely, as it's not adding a massive amount of value at this point.

Reply:
Yep, I came to that same conclusion after another look. There are only a few paths that reap the value of `SetBroadcastMessage`; many other places are manually adding the message reference in a different way.

Reply:
cool - resolving this for now, but we can come back to this if it confuses us again in the future